### PR TITLE
RA - Add Allies07

### PR DESCRIPTION
**Pull request by Smittytron**
My goal for difficulty here is to have an easy mode, a standard fight the enemy mode for Normal difficulty, and a bit of oomph for hard difficulty. (In the way of 30 tanks dropping on the player at once)

I've reassessed my former goal of adding a 'Real Tough Guy' fourth difficulty to each mission. I now believe that those only fit in a way where they completely alter how the mission is played. Allies02 and Allies05a are golden examples of how this should be done. I couldn't think of a way to change how this mission plays out, therefore, three difficulty levels.

### CODE REVIEW

**Review by Orb370 — DISMISSED** [review]
Played through the mission on normal difficulty with a few restarts. No issues.

File: mods/ra/maps/allies-07/allies07.lua [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		greece.MarkFailedObjective(CaptureRadarDomeObj)
+	end)
+
+	Trigger.OnCapture(RadarDome, function()
+		greece.MarkCompletedObjective(CaptureRadarDomeObj)
+		BaseRaids()
+	end)
+end
+
+BaseRaids = function()
```
Comment (Smittytron) [inline mods/ra/maps/allies-07/allies07.lua]: Wasn't sure about this one. Is there a way to write this?

**Review by SoScared — DISMISSED** [review]
Played Normal and Hard, encountered no issues.

I feel Normal might a bit too strong at the beginning, baiting v2 shots, heavy tanks, yaks and careful targeting. Could be a high barrier for some.

**Comment by Smittytron** [comment]
Updated. Removed v2 production from easy and normal, and renamed waypoints per IRC nit.

**Review by SoScared — DISMISSED** [review]
Mission replayed on Normal. Feels well paced for its difficulty level.

File: mods/ra/maps/allies-07/allies07.lua [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+				end)
+			end)
+		end)
+	end
+end
+
+StartTimerFunction = function()
+	if Map.LobbyOption("difficulty") == "hard" then
+		StartTimer = true
+	end
```
Comment (abcdefg30) [inline mods/ra/maps/allies-07/allies07.lua]: Can we get a "Timer started" notification here?

File: mods/ra/maps/allies-07/allies07.lua [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	{ HardEntry1.Location, HardLanding1.Location },
+	{ HardEntry2.Location, HardLanding2.Location },
+	{ HardEntry3.Location, HardLanding3.Location },
+	{ HardEntry4.Location, HardLanding4.Location },
+	{ HardEntry5.Location, HardLanding5.Location },
+	{ HardEntry6.Location, HardLanding6.Location }
+}
+
+SendArmoredBattalion = function()
+	Utils.Do(BattalionWays, function(way)
```
Comment (abcdefg30) [inline mods/ra/maps/allies-07/allies07.lua]: The "Enemy units approaching" speech notification would be nice here.

**Review by abcdefg30 — COMMENTED** [review]
Looks ok to me otherwise.

**Comment by Smittytron** [comment]
Updated

File: mods/ra/maps/allies-07/rules.yaml [hunk 4]
```diff
@@ -0,0 +1,133 @@
+Player:
+	PlayerResources:
+		DefaultCash: 4000
+
+World:
+	LuaScript:
+		Scripts: allies07.lua, allies07-AI.lua
+	MissionData:
+		Briefing: Briefing: LANDCOM 16 HQS.\nTOP SECRET.\nTO: FIELD COMMANDER A9\n\nINTERCEPTION OF SOVIET COMMUNIQUE INDICATES THEIR IRON CURTAIN RESEARCH WAS SET BACK BY ESPIONAGE. EXCELLENT WORK, COMMANDER!\n\nCOMMUNIQUE WAS TRACED BACK TO SECRET SOVIET BASE IN BORNHOLM. INVESTIGATE POSSIBLE CONNECTION WITH IRON CURTAIN RESEARCH. CAPTURE RADAR CENTER AND DESTROY SUB PRODUCTION CAPABILITY.\n\nCONFIRMATION CODE 1138.\n\nTRANSMISSION ENDS.
```
Comment (abcdefg30) [inline mods/ra/maps/allies-07/rules.yaml]: Duplicate `Briefing:`.
Please add a `\n` at the end, as well.

**Comment by Smittytron** [comment]
Updated

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/4df4317d3cfa0947c6510e1884f0d2b11e7c8ce1...d5913ed49be02ef26ec6a772051f6b87baade90e)

**Pull request merged**